src/ScrollSpy/ScrollSpy.tsx

```diff
@@ -159,26 +161,30 @@ const ScrollSpy = ({
             }
 
             prevIdTracker.current = changeHighlightedItemId;
-            window.history.pushState({}, "", `#${changeHighlightedItemId}`);
+            if (updateHistoryStack){
+              window.history.pushState({}, "", `#${changeHighlightedItemId}`);
+            }
```

Comment:
Downside of this is it will also not display the new hash in the address bar. Imo a better solution would be to use `window.history.replaceState` which will update the current page in history rather than adding a new one. That way we don't end up polluting the history with a bunch of new entries, but we also update the url in the address bar.

Reply:
I agree with @dslovinsky